### DISCUSSION TITLE
Clad failed to place the generated derivative in the object

### DISCUSSION THREAD

**Discussion by Monal-Patel** [vgvassilev/clad#878] [Q&A] [answered]
I am seeing "Clad failed to place the generated derivative in the object" for the following lines
```
  // Call clad to generate the derivative of f wrt x.
  auto f_dx = clad::differentiate(f, "x");
  // Execute the generated derivative function.
  std::cout << f_dx.execute(/*x=*/3, /*y=*/4) << std::endl;
  // Dump the generated derivative code to standard output.
  f_dx.dump();
```
I notice, in Differentiator.h lines 160-163.
>          // clad did not place the derivative in this object. This can happen
>          // upon error of if clad was disabled. Diagnose.
>          printf("clad failed to place the generated derivative in the object\n");
>          printf("Make sure calls to clad are within a #pragma clad ON region\n");

At the end of Differentiator.h file there is #pragma clad ON. So, if I include Differentiator.h file, which I do, I do not need the #pragma.

I also added 
```
#pragma clad on
// clad:: calls
#pragma clad off
```
however, this also does not fix the issue.

Any ideas of what could be happening here?

**Comment by vgvassilev** [comment]
Can you paste the command line you use?

**Comment by Monal-Patel** [comment, marked as answer]
I added clad in my makefile incorrectly,-fplugin=path/to/clad.so was not being being passed as an argument.
